src/processing/parsing/datasets.py

```diff
@@ -0,0 +1,124 @@
+from torch.utils.data.dataset import Dataset
+import torch
+import numpy as np
+import os
+import os.path as op
+import pickle
+
+NOTE_TICK_LENGTH = 37
+HARMONY_ROOT_LENGTH = 12
+HARMONY_PITCH_CLASSES_LENGTH = 12
+
+class MidiTicksDataset(Dataset):
+    """
+    This defines the loading and organizing of parsed MusicXML data into a MIDI tick database.
+    """
+
+    def __init__(self, load_dir, measures_per_seq=8, hop_size=4, target_type="full_sequence"):
+        """
+        Loads the MIDI tick information, groups into sequences based on measures.
+        :param load_dir: location of parsed MusicXML data
+        :param measures_per_seq: how many measures of ticks should be considered a sequence
+        :param hop_size: how many measures to move in time when creating sequences
+        :param target_type: if "full_sequence" the target is the full input sequence, if "next_step" the target is
+                the last tick of the input sequence
+        """
+
+        if not op.exists(load_dir):
+            raise Exception("Data directory does not exist.")
+
+        self.sequences = []
+        self.targets = []
+
+        for file in os.listdir(load_dir):
+            if op.splitext(file)[1] != ".pkl":
+                print("Skipping %s..." % file)
+                continue
+
+            song = pickle.load(open(op.join(load_dir, file), "rb"))
+
+            if song["metadata"]["ticks_per_measure"] != 96:
```

Comment:
this is actually not a great thing to do. Need to make a report, but it is possible that a song with 4/4 has less than 96 tick division for whatever reason.

Reply:
Hmm, I'm not really sure what you mean. Our parsing function ensures that everything it parses has 24 ticks/quarter note. It imposes that based on whatever divisions the XML is in, as long as it's greater than 24. Maybe that's a shortcoming of the parser, but anything going through our data processing pipeline that's in 4/4 will have 96 ticks per measure.